.github/workflows/ci.yml

```diff
@@ -267,22 +267,29 @@ jobs:
       - uses: software-mansion/setup-universal-sierra-compiler@v1
       - run: cargo test --release -p scarb-api
 
-  scarbfmt:
+  scarb-fmt-and-lint:
+    name: Format and lint packages
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@7b1c307e0dcbda6122208f10795a713336a9b35a
+        with:
+          toolchain: stable
+      - uses: Swatinem/rust-cache@82a92a6e8fbeee089604da2575dc567ae9ddeaab
       - uses: software-mansion/setup-scarb@v1
-      - name: Check cairo files format
+      - name: Build forge
+        run: cargo build --release -p forge
+      - name: Setup forge template packages
         run: |
-          output=$(find . -type f -name "Scarb.toml" -execdir sh -c '
-              echo "Running \"scarb fmt\" in directory: $PWD"
-              scarb fmt --check
-          ' \;)
-          echo "$output"
-          if grep -iq "Diff" <<< "$output"; then
-              exit 1
-          fi
-          exit 0
+          chmod +x ./scripts/setup_snforge_templates.sh
+          ./scripts/setup_snforge_templates.sh
+      - name: Check packages formatting
+        run: ./scripts/scarb_fmt.sh
+      - uses: software-mansion/setup-scarb@v1
+        with:
+          scarb-version: "2.10.0"
+      - name: Check packages linting
+        run: ./scripts/scarb_lint.sh
```

Comment:
note: `scarb lint` is available from 2.10.0, hence need to use different scarb version